src/Advection/weno_interpolants.jl

```diff
@@ -170,107 +171,86 @@ end
 
 # Smoothness indicators for stencil `stencil` for left and right biased reconstruction
 for buffer in [2, 3, 4, 5, 6]
-    @eval begin
-        @inline smoothness_sum(scheme::WENO{$buffer}, ψ, C) = @inbounds $(metaprogrammed_smoothness_sum(buffer))
-    end
-
-    for stencil in [0, 1, 2, 3, 4, 5]
-        @eval begin
-            @inline  left_biased_β(ψ, scheme::WENO{$buffer}, ::Val{$stencil}) = @inbounds smoothness_sum(scheme, ψ, smoothness_coefficients(scheme, Val($stencil)))
-            @inline right_biased_β(ψ, scheme::WENO{$buffer}, ::Val{$stencil}) = @inbounds smoothness_sum(scheme, ψ, smoothness_coefficients(scheme, Val($stencil)))
-        end
+    @eval @inline smoothness_sum(scheme::WENO{$buffer}, ψ, C) = @inbounds $(metaprogrammed_smoothness_sum(buffer))
+    
+    for stencil in 0:buffer-1
+        @eval @inline biased_β(ψ, scheme::WENO{$buffer, FT}, ::Val{$stencil}) where FT = 
+            smoothness_sum(scheme, ψ, FT.($(smoothness_coefficients(Val(buffer), Val(stencil)))))
```

Comment:
it may be better to use `map` vs broadcasting where this is easy, I think this could affect precompilation time